### ISSUE TITLE
Snippets selection checkboxes have no label for screen reader users

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#5328]
In the Snippets admin interface, it’s possible to select snippets for bulk actions by using checkboxes. Those checkboxes have no visible label, and no label for screen-reader users either. At the moment, they are read as "unticked tick box" and that’s it.

<img width="1065" alt="snippets-select-checkboxes" src="https://user-images.githubusercontent.com/877585/58497615-40829580-8174-11e9-888e-4e4be8403d2e.png">

---

There should be a label providing context for screen reader user, stating what the checkbox is for, as well as which item it applies to.

**Comment by thibaudcolas** [comment]
@martey to look at this.
